spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/test/EventuallyMatcher.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.deployer.spi.test;
+
+import org.hamcrest.Description;
+import org.hamcrest.DiagnosingMatcher;
+import org.hamcrest.Matcher;
+
+
+/**
+ * A matcher that will evaluate another matcher repeatedly until it matches, or fail after some number of attempts.
+ *
+ * @param <U> the type the wrapped matcher operates on
+ *
+ * @author Eric Bottard
+ */
+public class EventuallyMatcher<U> extends DiagnosingMatcher<U> {
+
+	private final Matcher<U> delegate;
+
+	private int noAttempts;
+
+	private int pause;
+
+	public EventuallyMatcher(Matcher<U> delegate) {
+		this(delegate, 20, 100);
+	}
+
+	public EventuallyMatcher(Matcher<U> delegate, int noAttempts, int pause) {
```

Comment:
I think `maxAttempts` would be more clear than `noAttempts` throughout (for one thing it IS a max... if the match occurs for any attempt less than the max it will not continue)